### PR TITLE
Fix some problems of thievery 

### LINKED ISSUE #1572: npc dupe and crash
### Describe the bug

eazy step to take every NPC's items, and sometimes crashes
when crash, it was written that the object was not found

### Steps To Reproduce

Dupe:
1) go to NPC's items
2) wait 1 move
3) take 1 item
*NPCs will not have time to react in time
4) drop this item
*a dialog box appears
5) say "a) i will drop this item"
6) dialog box close, item is your

How to crash:
1) go to NPC's items
2) wait 1 move
3) take 2 items
... i dont remember( , i test it on EMP grenades
4) ignore massage "NPC trying to talk to you"
*crash, writing that cant find item

### Expected behavior

Fix that NPCs immediately react to the theft

### Screenshots

_No response_

### Versions and configuration

i test it on win64(not msvc), but i think it works on another platforms

### Additional context

[debug.log](https://github.com/cataclysmbnteam/Cataclysm-BN/files/8773205/debug.log)
crash on 4th massage from up, i think

### LINKED ISSUE #1581: Problems of moving items that doesn't belong to the player faction
### Describe the bug

When you try to move any item that doesn't belong to you, if the item is count by charges, the charges simply disappear and no things moved, then if you try to move that item to your inventory, the game will crash.

### Steps To Reproduce

Described above.

### Expected behavior

Nothing should be moved and the charges should not change.

### Screenshots

GIF 2022-5-29 22-10-47

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 26c7ef1e612-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [dda],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    Elevated bridges [elevated_bridges]
]

### Additional context

Fixes is done. Just put the issue here. I'll open a PR later.

### PR DESCRIPTION
**Pull request by SaintCirno9**
#### Summary

SUMMARY: [Bugfixes] "Fix some problems of thievery "

#### Purpose of change

Fix #1572 
Fix #1581 
Action "Sort out my loot" does not move items that do not belong to the player faction (no theft).

#### Describe the solution

For #1572, make the NPC talk about thievery immediately once they notice you steal something.
For #1581, move the code to right position (I believe they are placed wrong during the migration).
Add items owner check in move loot action.

#### Testing

Tested thievery in refugee center. Works as intended apparently.

### CODE REVIEW

File: src/npc.h [hunk 1]
```diff
@@ -1341,6 +1340,7 @@ class npc : public player
         static constexpr tripoint_abs_omt no_goal_point{ tripoint_min };
         job_data job;
         time_point last_updated;
+        static bool has_thievery_witness;
```
Comment (Coolthulhu) [inline src/npc.h:1343]: `static` - now that's evil.
And unnecessary evil at that. You could inline `npc::witness_thievery()` where it is used and just handle all witnessing there.

Comment (SaintCirno9) [reply src/npc.h]: I‘ve noticed there are still some static variables in npc.h, so they are considered bad right?

**Comment by Coolthulhu** [comment]
Certainly looks better now, though I don't like it that `item` method sets player activity values.

Will have to spend some time to test it for edge cases, but if it works like it looks it does, it's good.

**Comment by Coolthulhu** [comment]
Can't move-steal the items in advanced inventory at all. I assume this is intended.
